editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/properties/AdvancedBeanPropertiesSection.java

```diff
@@ -313,4 +322,25 @@ public void widgetSelected(SelectionEvent event) {
 
 		});
 	}
+
+	private void createPublicStaticMethodBrowseButton(Composite composite, Text field) {
+		Button browseMethodButton = new Button(composite, SWT.PUSH);
+		browseMethodButton.setText("..."); //$NON-NLS-1$
+		browseMethodButton.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent event) {
+				Object control = modelMap.get(GlobalBeanEIP.PROP_CLASS);
```

Comment:
I feel that there is some way to factorize the code with the "createMethodBrowseButton" it seems that there is a single method call different